internal/templates/validate.go

```diff
@@ -0,0 +1,76 @@
+package templates
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"text/template"
+)
+
+// ValidateTemplate validates a text template results in valid JSON
+// when it's executed with empty template data. If template execution
+// results in invalid JSON, the template is invalid. When the template
+// is valid, it can be used safely. A valid template can still result
+// in invalid JSON when non-empty template data is provided.
+func ValidateTemplate(text string) error {
+	failMessage := "fail"
+	funcMap := GetFuncMap(&failMessage)
+
+	// prepare the template with our template functions
+	tmpl, err := template.New("template").Funcs(funcMap).Parse(text)
+	if err != nil {
+		return fmt.Errorf("error parsing template: %w", err)
+	}
+
+	// execute the template with empty data, resulting in nulls for fields
+	// that aren't filled in the template
+	buf := new(bytes.Buffer)
+	if err := tmpl.Execute(buf, nil); err != nil {
+		return fmt.Errorf("error validating template execution: %w", err)
+	}
+
+	// resulting JSON should be valid; if not, the template was not formatted correctly
+	if ok := json.Valid(buf.Bytes()); !ok {
+		// determine what's wrong with the JSON exactly; the `Valid` method doesn't return that
+		var m map[string]interface{}
+		if err := json.NewDecoder(buf).Decode(&m); err != nil {
+			return fmt.Errorf("invalid JSON: %w", enrichJSONError(err))
+		}
+	}
+
```

Comment:
After this morning discussions, perhaps the most sensible thing would be to just validate with the Parse.
```suggestion
```